### ISSUE TITLE
Top-level access to the loss function (For example, to add an ArcFace)

### ISSUE THREAD

**Issue by medphisiker** [ultralytics/ultralytics#7863]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

Hello,

I think it would be great to add some top-level access to the loss function to be able to modify it and replace it with your own.
Suppose we want to train a model for face identification and we would like to use ArcFace loss. Or perhaps we want to add an addition to the original loss function in the form of a ArcFace loss with some weight.

It seems that adding the possibility of modifying the loss function through the top-level api would expand the possibilities of adapting models to a wider range of tasks.

### Use case

Perhaps we want to train a face detection model to identify faces and want the model to use ArcFace loss during training.
Or maybe we have some kind of specific task for which we want to use a non-standard loss function.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by Kayzwer** [comment]
@medphisiker you probably need to take some time to go through the whole project, but path to loss functions is `/ultralytics//ultralytics/utils/loss.py`.
